### PR TITLE
Improved handling for mirroring errors

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Unless this is a trivial change, please include a CHANGELOG entry.<br />
Run <code>osc vc</code> in the <code>package</code> directory to add one.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by thutterer** [comment]
> Unless this is a trivial change, please include a CHANGELOG entry.

:grin::point_up:

**Comment by ikapelyukhin** [comment]
@thutterer done! :grimacing::+1:

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The version of RMT is not consistent. These files must specify the same version:</p>
<ul>
  <li><code>lib/rmt.rb</code> specifies <code>1.0.2</code></li>
  <li><code>Makefile</code> specifies <code>1.0.1</code></li>
  <li><code>package/rmt-server.spec</code> specifies <code>1.0.2</code></li>
<</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: spec/lib/rmt/mirror_spec.rb [hunk 1]
```diff
@@ -402,4 +417,21 @@
...
+    subject(:replace_directory) { rmt_mirror.send(:replace_directory, 'src', 'dst') }
+
+    let(:rmt_mirror) do
+      described_class.new(
+        mirroring_base_dir: @tmp_dir,
+        logger: logger,
+        mirror_src: false
+      )
+    end
+
```
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply spec/lib/rmt/mirror_spec.rb]: Yeah, sure. I'm not sure why we never had a test for this specific method before.

File: lib/rmt/mirror.rb [hunk 2]
```diff
@@ -107,20 +99,18 @@ def mirror_license
...
-      end
-    rescue RMT::Downloader::Exception => e
-      raise RMT::Mirror::Exception.new("Error during mirroring metadata: #{e.message}")
+    File.open(directory_yast).each_line do |filename|
+      filename.strip!
+      next if filename == 'directory.yast'
+      @downloader.download(filename)
     end
+  rescue RMT::Downloader::Exception => e
+    raise RMT::Mirror::Exception.new("Error during mirroring license: #{e.message}")
```
[the remark this answers is not in the document]
Comment (ikapelyukhin) [reply lib/rmt/mirror.rb]: Done.

**Pull request merged**